packages/api-server/api_server/dependencies.py

```diff
@@ -10,7 +10,7 @@
 
 
 def pagination_query(
-    limit: int | None = Query(None, gt=0, le=1000, description="defaults to 100"),
+    limit: int | None = Query(None, gt=0, le=5000, description="defaults to 100"),
```

Comment:
tbh 1000 is already pushing it, what is the reason for such a high limit?

Reply:
there may be more than 30 scheduled tasks a day for the ongoing deployment, and we are required to be able to export a month's worth of tasks, so 1000 won't be enough. I set an arbitrary 5000 to just cover more ground.
What about 2000?

Reply:
> there may be more than 30 scheduled tasks a day for the ongoing deployment, and we are required to be able to export a month's worth of tasks, so 1000 won't be enough. I set an arbitrary 5000 to just cover more ground. What about 2000?

The limit shouldn't be an issue, in fact we should reduce it to say 100, and default to around 20. The frontend can/should do pagination to retrieve as many tasks as needed.

Reply:
Hmm, so in the case of trying to dump all the task states from DB, we should have the frontend query for 1-100, 101-200, 201-300 and so on, with separate calls?

Reply:
I've added endpoints for counting and made the frontend query by chunks

Also added a global loading backdrop to let users know that something is happening

Reply:
revert the limit to 1000.